modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/QuantumToolkit.java

```diff
@@ -445,6 +445,23 @@ public static <T> T runWithRenderLock(Supplier<T> supplier) {
         }
     }
 
+    public static void runInRenderThreadAndWait(Runnable runnable) {
+        try {
+            CountDownLatch latch = new CountDownLatch(1);
+            QuantumRenderer.getInstance().execute(() -> {
+                try {
+                    runnable.run();
+                } catch (Throwable e) {
+                    e.printStackTrace();
+                }
+                latch.countDown();
+            });
+            latch.await();
+        } catch (Throwable e) {
```

Comment:
Not sure whether we need to catch `Throwable` here, but we use that pattern all over the place in this file, so it can be reexamined as part of the cleanup bug I will file.